index/secondary_index_manager.hh

```diff
@@ -103,10 +105,19 @@ public:
     virtual void validate(const schema &schema, cql3::statements::index_prop_defs &properties, const std::vector<::shared_ptr<cql3::statements::index_target>> &targets, const gms::feature_service& fs) = 0;
 };
 
+struct index_metrics {
+    seastar::metrics::metric_groups metrics;
+    sstring ks_name;
```

Comment:
No need to store the ks_name or the idx_name inside the struct.
Create a local label in the file and use it in the constructor; there is no need to create one in the struct.